### PR TITLE
[AzureMonitorExporter] connection string via environment variable

### PR DESCRIPTION
**Pull request by TimothyMothra**
## Changes
- add support for Connection String via Environment Variable.
- refactor Transmitter ctor.
  this was getting large. I moved individual items into a static Initialize method.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs [hunk 1]
```diff
@@ -180,6 +143,80 @@ private static ExportResult IsSuccess(HttpMessage httpMessage)
...
+                    return null;
+                }
+            }
+
+            return null;
+        }
+
+        private static ConnectionVars InitializeConnectionVars(AzureMonitorExporterOptions options)
+        {
+            if (options.ConnectionString == null)
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:203]: Debatable topic :blush:

Which should take precedence - options set in the code or environment variable?

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:203]: The decision we made for AI SDK was anything hard-coded would take precedence over the env var.

**Review by vishweshbankwar — APPROVED** [review]
LGTM - a todo note to discuss debatable topic would be great and a test :)

**Comment by TimothyMothra** [comment]
> LGTM - a todo note to discuss debatable topic would be great and a test :)

Tests with Environment Variables is tricky and can have side effects. In Application Insights we solved this with an abstraction layer. 
Long term, I'd like to explore something similar here.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:203]: Adding @vishweshbankwar to this thread.
I'm not certain that there's anything to debate here. :)
I feel like this was settled when we built Connection Strings a few years ago.

If we need to debate this, I suggest we bring this up in the next Contributors Meeting and let all languages weigh in.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:203]: question: how does OTel SDK handles it? I cannot recall. I think we just need to be consistent with it.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:203]: > If we need to debate this, I suggest we bring this up in the next Contributors Meeting and let all languages weigh in. 👍

Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzureMonitorTransmitter.cs:203]: Added this to the agenda for next week.

**Pull request merged**